modules/avatar/react/lib/Avatar.tsx

```diff
@@ -10,80 +11,136 @@ export enum AvatarVariant {
   Dark,
 }
 
-export interface AvatarLocalProps {
+export interface AvatarProps extends React.HTMLAttributes<HTMLButtonElement> {
   /**
    * The variant of the Avatar default state. Accepts `Light` or `Dark`.
    * @default AvatarVariant.Light
    */
-  variant: AvatarVariant;
+  variant?: AvatarVariant;
   /**
    * The size of the Avatar.
    * @default SystemIconCircleSize.m
    */
-  size: SystemIconCircleSize | number;
+  size?: SystemIconCircleSize | number;
   /**
    * The alt text of the Avatar image.
    * @default Avatar
    */
-  altText: string;
+  altText?: string;
   /**
    * The url of the Avatar image.
    */
   url?: string;
+  /**
+   * The alternative container type for the button. Uses Emotion's special `as` prop.
+   * Will render an `button` tag instead of a `div` when defined.
+   */
+  as?: 'div';
+  /**
+   * The function called when the Avatar is clicked.
+   */
+  onClick?: React.MouseEventHandler<HTMLButtonElement>;
+  /**
+   * The ref to the button that the styled component renders.
+   */
+  buttonRef?: React.Ref<HTMLButtonElement>;
 }
 
-export interface AvatarProps extends AvatarLocalProps, React.HTMLAttributes<HTMLDivElement> {}
-
-export const avatarStyles: CSSObject = {
-  background: colors.soap200,
-  display: 'flex',
-  alignItems: 'center',
-  justifyContent: 'center',
-  padding: 0,
-  border: 0,
-  borderRadius: borderRadius.circle,
-  boxSizing: 'border-box',
-  overflow: 'hidden',
-  '& img': {
-    width: '100%',
-    height: '100%',
+const StyledContainer = styled('button', {
+  shouldForwardProp: prop => isPropValid(prop) && prop !== 'size',
+})<Pick<AvatarProps, 'variant' | 'size' | 'onClick'>>(
+  {
+    background: colors.soap200,
+    position: 'relative',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    padding: 0,
+    border: 0,
+    boxSizing: 'border-box',
+    overflow: 'hidden',
+    borderRadius: borderRadius.circle,
   },
-};
+  ({size}) => ({
+    height: size,
+    width: size,
+  }),
+  ({variant, onClick}) => ({
+    cursor: onClick ? 'pointer' : 'default',
+    '&:not([disabled])': {
+      '&:focus': {
+        outline: 'none',
+        ...(variant === AvatarVariant.Dark ? focusRing(2, 2) : focusRing(2)),
+      },
+    },
+    ...hideMouseFocus,
+  })
+);
 
-const StyledContainer = styled('div', {
-  shouldForwardProp: prop => isPropValid(prop) && prop !== 'size',
-})<Pick<AvatarProps, 'variant' | 'size'>>(
+const StyledStack = styled('span')<Pick<AvatarProps, 'size'>>(
   {
-    ...avatarStyles,
+    position: 'absolute',
   },
   ({size}) => ({
     height: size,
     width: size,
   })
 );
 
-export default class Avatar extends React.Component<AvatarProps> {
-  static Variant = AvatarVariant;
-  static Size = SystemIconCircleSize;
+const StyledImage = styled('img')<{isLoaded: boolean}>(
+  {
+    width: '100%',
+    height: '100%',
+    borderRadius: borderRadius.circle,
+    transition: 'opacity 200ms linear',
```

Comment:
Nit: Can we try 150ms? it feels a little slow atm

Reply:
Also, this is going to break our snapshot tests 🤔

Reply:
I will remove for now, we can think of a better way to handle this later

Reply:
👍 how do you deal with animations and transitions usually? Is there a class that overrides things like this for testing?

Reply:
Everything in the past has been tied to popups and there is a `transformOrigin` field. We support `transformOrigin={null}`, which removes the animation. Not sure how we'd want to do it here 🤔

Reply:
We could override the css, but snaps are done in storybook too right? Is there a way to conditional pass something in just for tests and not for a normal storybook run?

Reply:
I just realized that since we switched to opt-in snapshots (to keep our chromatic costs down), this component isn't captured anymore. We're going to be re-evaluating our storybook approach in the near-ish future as we continue work on a documentation site, so I wouldn't worry about this for now.